src/ApiClient.js

```diff
@@ -0,0 +1,71 @@
+// This class is a reference implementation. If you desire additional
+// functionality, you can work with the config options here, or copy this code
+// into your project and customize the ApiClient to your needs.
+
+class ApiClient {
+  constructor(passedConfig) {
+    const defaultConfig = {
+      methods: ['get', 'post', 'put', 'patch', 'delete'],
+      credentials: 'same-origin',
+      defaultHeaders: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+      },
+      format: 'json' // json, blob, text, arrayBuffer, ...
+    }
+
+    const config = {
+      ...defaultConfig,
+      ...passedConfig
+    }
+
+    if (!config.basePath) {
+      // e.g. 'https://example.com/api/v3'
+      throw new Error('You must pass a base path to the ApiClient')
+    }
+
+    config.methods.forEach(method => {
+      this[method] = (path, { params, data, otherConfig } = {}) => {
+        const queryString = this.queryString(params)
+        const { basePath, defaultHeaders, format, ...fetchConfig } = config
+        const headers = {
+          this.getHeaders(defaultHeaders, data),
```

Comment:
I don't understand why `data` is dumped into request headers. I expected that `data` would be given as a `body` property in the fetch config.